File: Analisis_data_shopee/shopee/umkm2.r
library("readxl")
library(openxlsx)
library(dplyr)
library(ggplot2)


df <- read_excel('umkm.xlsx')
# df <- subset(df, select=-No.)
df <- select(df, -No.)

# df$Nilai <- as.numeric(gsub(',', '.',df$Nilai))
df <- mutate(df, Nilai = as.numeric(gsub(',', '.',df$Nilai)))
colnames(df) <- c('Nama_Data', 'Nilai')
print(df)

# Menyiapkan data
x <- df$Nama_Data
y <- df$Nilai



# Membuat plot bar horizontal
bp <- barplot(df$Nilai, 
        # horiz = TRUE, 
        names.arg = df$Nama_Data, 
        col = "skyblue", 
        main = "Preferensi Pembelian Produk UMKM", 
        xlab = "Persentase (%)", 
        cex.names = 0.8)

# Menambahkan label angka di tengah bar
for (i in 1:nrow(df)) {
    text(bp[i], df$Nilai[i], labels = paste0(round(df$Nilai[i], 2), "%"), pos = 3, cex = 0.8, xpd = TRUE)
}

File: Analisis_data_shopee/shopee/umkm2.py
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_excel('umkm.xlsx')
df = df.drop('No.', axis=1)
df['Nilai'] = df['Nilai'].str.replace(",", ".").astype(float)
print(df)
print(df.dtypes)

# Membuat subplots
fig, ax = plt.subplots()
# Membuat horizontal bar plot
bars = ax.bar(df['Nama Data'], df['Nilai'], color='skyblue')
# Menetapkan judul plot
ax.set_title('Preferensi Pembelian Produk UMKM')
# Menetapkan label sumbu x
ax.set_xlabel('Persentase (%)')
# Rotasi label sumbu x agar lebih mudah dibaca
plt.xticks(rotation=45, ha='right')


# Menambahkan label angka di ujung bar
for bar in bars:
    height = bar.get_height()
    ax.annotate('{}'.format(round(height, 2)),
                xy=(bar.get_x() + bar.get_width() / 2, height),
                xytext=(0, 3),  # 3 points vertical offset
                textcoords="offset points",
                ha='center', va='bottom')

# Menampilkan plot
plt.show()
